### PR TITLE
Switch shoot.Status.SeedName at start of the reconcile process

### PR DESCRIPTION
**Pull request by krgostev**
**How to categorize this PR?**

/area control-plane-migration
/kind enhancement

**What this PR does / why we need it**:
This PR sets the correct `shoot.Status.SeedName` at the start of the  shoot operation process. This in case there was CPM process where the `spec.SeedName` is different from the `shoot.Status.SeedName`. The PR also adds description messages for operation `Migrate` and operation `Restore`.
**Which issue(s) this PR fixes**:
Fixes 

**Special notes for your reviewer**:
Part of #1631 
**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by timebertt** [comment]
/hold
until the `v1.19.0` release is cut.

File: pkg/gardenlet/controller/shoot/shoot_control.go [hunk 1]
```diff
@@ -569,6 +577,10 @@ func (c *Controller) updateShootStatusOperationStart(ctx context.Context, g gard
 				shoot.Status.TechnicalID = shootNamespace
 			}
 
+			if shoot.Status.SeedName != shoot.Spec.SeedName && operationType != gardencorev1beta1.LastOperationTypeMigrate {
```
Comment (rfranzke) [inline pkg/gardenlet/controller/shoot/shoot_control.go]:
```suggestion
			if !apiequality.Semantic.DeepEqual(shoot.Status.SeedName, shoot.Spec.SeedName) && operationType != gardencorev1beta1.LastOperationTypeMigrate {
```

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**